step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/functions/TokenForecastingExecutionPlugin.java

```diff
@@ -63,6 +64,14 @@ public void executionStart(ExecutionContext context) {
 
         // Token forecasting is always calculated, even if the agent provisioning is disabled
         TokenForecastingContext tokenForecastingContext = new TokenForecastingContext(availableAgentPools);
+        String executionThreadsAuto = context.getVariablesManager().getVariableAsString(ThreadPool.EXECUTION_THREADS_AUTO, null);
+        if (executionThreadsAuto != null && !executionThreadsAuto.isEmpty()) {
+            try {
+                tokenForecastingContext.setMaxParallelism(Integer.parseInt(executionThreadsAuto));
+            } catch (NumberFormatException e) {
+                logger.warn("Ignoring unparseable value for parameter {}: {}", ThreadPool.EXECUTION_THREADS_AUTO, executionThreadsAuto, e);
```

Comment:
Not sure if it wouldn't be better to throw an error in this case

Reply:
Get it. In that case it would be cleaner to validate this value at the beginning of the execution and report an execution lifecycle error. This could be done in another ticket too. So nothing to be changed here. I would however log it as ERROR

Reply:
Throwing error as requested